pkg/client/chartmuseum/chartmuseum_test.go

```diff
@@ -0,0 +1,233 @@
+package chartmuseum
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+	"reflect"
+	"sort"
+	"strings"
+	"testing"
+
+	"github.com/bitnami-labs/charts-syncer/api"
+	"github.com/bitnami-labs/charts-syncer/internal/cache"
+	"github.com/bitnami-labs/charts-syncer/internal/chartrepotest"
+	"github.com/bitnami-labs/charts-syncer/internal/utils"
+	"github.com/bitnami-labs/charts-syncer/pkg/client/types"
+	"helm.sh/helm/v3/pkg/time"
+)
+
+var (
+	cmRepo = &api.Repo{
+		Kind: api.Kind_CHARTMUSEUM,
+		Auth: &api.Auth{
+			Username: "user",
+			Password: "password",
+		},
+	}
+)
+
+func prepareTest(t *testing.T) (*Repo, error) {
+	t.Helper()
+
+	// Create temp folder and copy index.yaml
+	dstTmp, err := ioutil.TempDir("", "charts-syncer-tests-index-fake")
+	if err != nil {
+		t.Fatalf("error creating temporary folder: %v", err)
+	}
+	t.Cleanup(func() { os.RemoveAll(dstTmp) })
+	dstIndex := filepath.Join(dstTmp, "index.yaml")
+	if err := utils.CopyFile(dstIndex, "../../../testdata/index.yaml"); err != nil {
+		t.Fatal(err)
+	}
+
+	// Create tester
+	tester, cleanup, err := NewTester(t, cmRepo, false, dstIndex)
+	t.Cleanup(func() { cleanup() })
```

Comment:
if `cleanup` matches the `Cleanup` interface (`func()`):
```suggestion
	t.Cleanup(cleanup)
```